client/main.lua

```diff
@@ -419,19 +414,19 @@ end
 ---@param shop string
 local function startTestDriveTimer(time, shop)
     local gameTimer = GetGameTimer()
+    local timeMs = time * 1000
+
     CreateThread(function()
         while InTestDrive do
-            if GetGameTimer() < gameTimer + tonumber(1000 * time) then
-                local secondsLeft = GetGameTimer() - gameTimer
-                if secondsLeft >= tonumber(1000 * time) - 20 then
+            local currentGameTime = GetGameTimer()
+            local secondsLeft = currentGameTime - gameTimer
+            if currentGameTime < gameTimer + timeMs then
+                if secondsLeft >= timeMs - 50 then
```

Comment:
Why the change from -20 to -50? Also did you end up trying the condition where secondsLeft <= 0 instead?

Reply:
Yes, the secondsLeft timer count is upwards, not down, so if you use the suggested condition your car will immediately despawn. I changed the number as it feels to me like a little network delay on some users (since it doesn't happen in any of my two local computers), but I do see it on some users that connect remotely.